sdk/go/examples_test.go

```diff
@@ -0,0 +1,259 @@
+
+	dir := client.Directory().
+		WithNewFile("hello.txt", dagger.DirectoryWithNewFileOpts{
+			Contents: "Hello, world!",
+		}).
+		WithNewFile("goodbye.txt", dagger.DirectoryWithNewFileOpts{
+			Contents: "Goodbye, world!",
+		})
+
+	dirID, err := dir.ID(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	container := client.Container().From("alpine:3.16.2")
+
+	container = container.WithMountedDirectory("/mnt", dirID)
+
+	out, err := container.Exec(dagger.ContainerExecOpts{
+		Args: []string{"ls", "/mnt"},
+	}).Stdout().Contents(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("%q", out)
+
+	// Output: "goodbye.txt\nhello.txt\n"
+}
+
+func ExampleContainer_WithMountedCache() {
+	ctx := context.Background()
+	client, err := dagger.Connect(ctx)
+	if err != nil {
+		panic(err)
+	}
+	defer client.Close()
+
+	cacheKey := "example-" + time.Now().Format(time.RFC3339)
+
+	cacheID, err := client.CacheVolume(cacheKey).ID(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	container := client.Container().From("alpine:3.16.2")
+
+	container = container.WithMountedCache(cacheID, "/cache")
+
+	var out string
+	for i := 0; i < 5; i++ {
+		out, err = container.Exec(dagger.ContainerExecOpts{
+			Args: []string{
+				"sh", "-c",
+				"echo $0 >> /cache/x.txt; cat /cache/x.txt",
+				strconv.Itoa(i),
+			},
+		}).Stdout().Contents(ctx)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	fmt.Printf("%q", out)
+
+	// Output: "0\n1\n2\n3\n4\n"
+}
+
+func ExampleDirectory() {
+	ctx := context.Background()
+	client, err := dagger.Connect(ctx)
+	if err != nil {
+		panic(err)
+	}
+	defer client.Close()
+
+	dir := client.Directory().
+		WithNewFile("hello.txt", dagger.DirectoryWithNewFileOpts{
+			Contents: "Hello, world!",
+		}).
+		WithNewFile("goodbye.txt", dagger.DirectoryWithNewFileOpts{
+			Contents: "Goodbye, world!",
+		})
+
+	entries, err := dir.Entries(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(entries)
+
+	// Output: [goodbye.txt hello.txt]
+}
+
+func ExampleHost_Workdir() {
+	ctx := context.Background()
+	client, err := dagger.Connect(ctx, dagger.WithWorkdir("."))
+	if err != nil {
+		panic(err)
+	}
+	defer client.Close()
+
+	readme, err := client.Host().Workdir().Read().File("README.md").Contents(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	lines := strings.Split(strings.TrimSpace(readme), "\n")
+	fmt.Println(lines[0])
+
+	// Output: # Dagger Go SDK
+}
+
+func ExampleHost_EnvVariable() {
+	ctx := context.Background()
+	client, err := dagger.Connect(ctx)
+	if err != nil {
+		panic(err)
+	}
+	defer client.Close()
+
+	os.Setenv("SEKRIT", "hunter2")
+
+	secretID, err := client.Host().EnvVariable("SEKRIT").Secret().ID(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	alpine := client.Container().From("alpine:3.16.2")
+	leaked, err := alpine.
+		WithSecretVariable("PASSWORD", secretID).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"sh", "-c", "echo $PASSWORD"},
+		}).
+		Stdout().Contents(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(leaked)
```

Comment:
I suspect this example will cause confusion to those not familiar with secrets already (why am I printing my secret password?). I'd probably just leave it out for now until we have a better story around secrets.